components/layout_2020/positioned.rs

```diff
@@ -928,9 +928,13 @@ impl<'a> AbsoluteAxisSolver<'a> {
             _ => return None,
         };
 
+        let default_overflow_rect = (alignment_container.origin.max(Au::zero())) +
+            (alignment_container.length.max(Au::zero()));
+
         let mut value_after_safety = self.alignment.value();
         if self.alignment.flags() == AlignFlags::SAFE &&
-            margin_box_axis.length > alignment_container.length
+            (margin_box_axis.length > alignment_container.length ||
+                alignment_container.origin < Au::zero())
```

Comment:
Don't force start alignment just because the start inset is negative.

```html
<!DOCTYPE html>
<div style="margin: 40px; border: solid; position: relative; width: 100px; height: 100px">
  <div style="position: absolute; z-index: -1; inset: -20px; background: cyan; place-self: safe end; width: 50px; height: 50px"></div>
</div>
```